src/components/BookDetails/BookDetails.js

```diff
@@ -0,0 +1,44 @@
+import { Link, useParams } from "react-router-dom";
+import "./BookDetails.css";
+import { useDispatch, useSelector } from "react-redux";
+import Nav from "../Navigation/Navigation";
+import { actions } from "../../BookReducer";
+
+export default function BookDetails() {
+  const { id } = useParams();
+  const card_dispatcher = useDispatch();
+  const buyBtn_dispatcher = useDispatch();
+  const book = useSelector((state) => state.bookReducer.books[id - 1]);
+  function handleCartClick() {
+    card_dispatcher(actions.addToCart(book));
+  }
+  function handleBuyBtnClick() {
+    buyBtn_dispatcher(actions.addToCart(book));
+  }
+
+  return (
+    <>
+      <Nav title={book.title} />
```

Comment:
Do not render Nav in each of the route explicitly, add the Nav component at the root component (App)